test/mpi/threads/rma/mt_putget.c

```diff
@@ -0,0 +1,282 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2018 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2018 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+
+/* Multi-thread One-Sided MPI Put/Get Test
+ *
+ * This code performs several put/get operations into a 2D array using
+ * multiple threads. The array has dimensions [X, Y] and the transfer is
+ * done by NTHREADS threads concurrently. Both small msg and huge msg transfers
+ * are tested.
+ */
```

Comment:
General question for all tests: why do we have different sync type only in the `putget` test ? The framework is a little unclear to me. Can we test all RMA operations, with all sync types, with both `win_create` and `win_allocate` (they internally use different op issuing routine) ? 

Moreover, we'd like separate each test unit into different test binary so each test can finish in short time. The source file can be the same as the test logic is the same.